libraries/botbuilder-dialogs-adaptive/src/actions/httpRequest.ts

```diff
@@ -137,20 +144,40 @@ export class HttpRequest<O extends object = {}> extends Dialog<O> {
         /**
          * TODO: replace the key value pair in json recursively
          */
-        const url = this.url.evaluate(this.id, dc.state);
-        const body = this.body.evaluate(this.id, dc.state);
-        const headers = this.headers.evaluate(this.id, dc.state);
 
-        const response = await fetch(url, {
-            method: this.method.toString(),
-            headers: headers,
-            body: body
-        });
+        const url = stringTemplate.format(this.url, dc);
```

Comment:
>const url = stringTemplate.format(this.url, dc); [](start = 8, length = 48)

This is likely a bug in the C# version of the code as well, but we need to make sure we URL encode any query string params.  We should decide on a solution for both JS and C# here and fix in both.

Reply:
> > const url = stringTemplate.format(this.url, dc); [](start = 8, length = 48)
> 
> This is likely a bug in the C# version of the code as well, but we need to make sure we URL encode any query string params. We should decide on a solution for both JS and C# here and fix in both.

Do we need to replace all string templates with expression property?